reconcile/utils/oc.py

```diff
@@ -690,57 +690,66 @@ def get_obj_root_owner(self, ns, obj, allow_not_found=False,
                     )
         return obj
 
-    @staticmethod
-    def secret_used_in_pod(name, pod):
-        volumes = pod['spec']['volumes']
-        for v in volumes:
-            volume_item = v.get('secret', {})
-            try:
-                if volume_item['secretName'] == name:
-                    return True
-            except KeyError:
-                continue
-        containers = pod['spec']['containers']
-        for c in containers:
-            for e in c.get('envFrom', []):
-                try:
-                    if e['secretRef']['name'] == name:
-                        return True
-                except KeyError:
-                    continue
-            for e in c.get('env', []):
-                try:
-                    if e['valueFrom']['secretKeyRef']['name'] == name:
-                        return True
-                except KeyError:
-                    continue
-        return False
+    def secret_used_in_pod(self, name, pod):
+        used_resources = self.get_resources_used_in_pod_spec(pod["spec"], "Secret")
+        return name in used_resources
+
+    def configmap_used_in_pod(self, name, pod):
+        used_resources = self.get_resources_used_in_pod_spec(pod["spec"], "ConfigMap")
+        return name in used_resources
 
     @staticmethod
-    def configmap_used_in_pod(name, pod):
-        volumes = pod['spec']['volumes']
-        for v in volumes:
-            volume_item = v.get('configMap', {})
-            try:
-                if volume_item['name'] == name:
-                    return True
-            except KeyError:
-                continue
-        containers = pod['spec']['containers']
-        for c in containers:
-            for e in c.get('envFrom', []):
-                try:
-                    if e['configMapRef']['name'] == name:
-                        return True
-                except KeyError:
-                    continue
-            for e in c.get('env', []):
+    def get_resources_used_in_pod_spec(spec: Dict[str, Any], kind: str) -> Dict[str, Set[str]]:
+        if kind not in ("Secret", "ConfigMap"):
+            raise KeyError(f"unsupported kind for get_resources_used_in_pod_spec: {kind}")
```

Comment:
If this function changes its name you'll have to adapt the exception string too. Since you are going to have it in the stack trace anyways, why not either refer to the objective of the function or do away with it entirely?

```suggestion
            raise KeyError(f"unsupported resource kind {kind}")
```

(you could use `inspect.currentframe().f_code.co_name` but I doubt the extra import helps much here)